### PR TITLE
 feat(config): use separate defaults for all states

### PR DESCRIPTION
**Pull request by ErisDS**
At the moment this is untested and pure opinion - again happy to do more work if other people agree.

IMO it only makes sense that the defaults are shared, whilst the default templates are in use to reduce duplication.

As soon as you override one of the shared defaults (nginx.conf or server.conf) unexpected things happen:

e.g
- I specify a custom hard-coded nginx.conf file that I use on all servers. I don't expect it to suddenly be impossible to configure passenger.
- Similarly, if I do something custom with server.conf, I don't expect that snippers are affected`

This can currently be worked around by specifying your own names for files in tofs.source_files, so the change may be undesirable.

This tripped me up so thought I'd raise to bring it to attention of maintainers.

### CODE REVIEW

**Comment by ErisDS** [comment]
I take it back, this cannot be worked around. See the context in slack here: https://saltstackcommunity.slack.com/team/ULFR9ES64

TL;DR if you override the template name with source files, it completely overrides the file switch.

Example here: https://github.com/TryGhost/nginx-formula/blob/e957a022763db369dece479479301edaa7223c91/nginx/servers_config.sls#L116

I expected that overriding `source_files` would change the cascade from looking for files with the dynamic server name, and then `server.conf` to looking for files the dynamic server name, and then my configured value in source files.

E.g `server.conf` is replaced with whatever `source_files.server_conf_file_managed` is configured to.

Instead, it gets rid of the ability to look for the dynamic server name, and only looks for the hardcoded `source_files.server_conf_file_managed` value.

**Comment by ErisDS** [comment]
Bottom line - the reuse of default templates for multiple states is a severe limitation, you cannot override one default without changing the behaviour of another state.

There is no workaround or fix. Meaning you effectively cannot customise the defaults in any way that is intended to tailor them to one state or another - a severe limitation.

**Comment by myii** [comment]
As discussed in Slack, merging https://github.com/saltstack-formulas/template-formula/pull/152 (and propagating to all formulas) will alleviate these problems.

**Comment by ErisDS** [comment]
Yes https://github.com/saltstack-formulas/template-formula/pull/152 will mean there is at least a workaround. However, it would be ideal if no workaround was necessary :)

**Comment by myii** [comment]
The merge of #248 has brought in all the TOFS changes as mentioned [above](https://github.com/saltstack-formulas/nginx-formula/pull/247#issuecomment-514279898).

**Comment by aboe76** [comment]
@ErisDS can you please take a look and rebase your pr?
